### ISSUE TITLE
Awesome work, roadmap+request

### ISSUE THREAD

**Issue by TimNZ** [komarovalexander/ka-table#154]
Love this library, thanks so much for making it and being free.

It's the only one that starts matching the quality of commercial 
https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/getting-started/ 

I'm stealing nice features from the Devexpress grid, and want to implement summary rows for the table, and groups.
https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/summary-row/

Hooks into the rendering pipeline before/pre and after/post <group/data> row creation would be great. 
The hooks would optionally return new rows to be inserted before/after the current row being processed.

Skimmed through the code and can't see an obvious way of doing that now.

**Comment by komarovalexander** [comment]
Hi @TimNZ Thanks for feedback! I will investigate your proposal next week! I will let you know if any updates

**Comment by TimNZ** [comment]
@komarovalexander lol I see you used to work at Devexress :)

**Comment by TimNZ** [comment]
@komarovalexander I'm scoping up dev for what feels like more complete grouping functionality, and so far:

**support a dynamic value mechanism for groups, overriding raw group column value**
e.g. I want to group by rows having an age in a range e.g. '1 - 10 years old'
Either a new prop on the Group type passed in groups: [] table prop, such as 'value' that supports a function,
or a 'groupValue' childComponent mechanism to be consistent with current approach, though childComponents only appears to be Ui/displayed content.

```
tableProps = { groups: [{columnKey: 'age', value: ({data}) => { ... switch(ageBracket) { case '1_10': return  '1 - 10 years old'} } } ] }
```
or 
```
<Table
        {...tableProps}
        childComponents={{
					groupValue: { }
         }}
</Table>
```

**add 'value' prop to IGroupRowProps, to use as alternative to the formatted 'text'**

**add 'groupFooterRow' component**
If provided, rendered after all group data rows - prob at least same props as IGroupRowProps

**for consistency with groupFooterRow, consider migrating groupRow -> groupHeaderRow**

**add 'tableFooterRow' and 'tableFooterCell' component**
These could be regarded as table level 'groupXXXXXRow' components, with a null group key, for sharing/consistency

Thoughts?

From this, Devexpress summary support could be straight forward to add as an addon custom childComponent, vs baking into KA table core.
Might need to pass in more table definition data into the IXXXXXXProps so custom funcs have more info to work with without needing to know things externally.


Similar  to what https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/grouping/ offers.

**Comment by komarovalexander** [comment]
> support a dynamic value mechanism for groups, overriding raw group column value
 
at this time it could be done this way https://stackblitz.com/edit/table-grouping-custom-cell-ts-zr8ejj?file=Demo.tsx

```
      childComponents={{
        groupCell: {
          content: (props) => {
            switch(props.groupKey[0]) { 
              case '1-10': return  '1 - 10 years old';
              case '11-20': return  '11 - 20 years old';
            }
          }
        }
      }}
```

probably format should work in case of grouping, this is a good catch
```
      // format: (value) => {
      //  return 'custom text';
      // },
```
will investigate it and **`format` will be fixed soon** - estimated time: end of this week
also **`value` should be added to `props` in the case of groupCell content**  - estimated time: end of this week


as for 
> add 'groupFooterRow' component
> If provided, rendered after all group data rows - prob at least same props as IGroupRowProps
> for consistency with groupFooterRow, consider migrating groupRow -> groupHeaderRow
> add 'tableFooterRow' and 'tableFooterCell' component

This is a nice idea, I need some time to investigate how to implement it in ka-table. But these are great candidates to be added in the next releases. I will let you updates regarding it probably in next 2 weeks

**Comment by komarovalexander** [comment]
group formatting in 6.8.3: https://stackblitz.com/edit/table-grouping-custom-cell-ts-wzztqj?file=Demo.tsx

**Comment by TimNZ** [comment]
Awesome, thanks!

**Comment by komarovalexander** [comment]
Hi @TimNZ 
The summary row has just released. It is available in v6.9.0: https://komarovalexander.github.io/ka-table/#/summary

Summary row for groups will be released later this month

**Comment by TimNZ** [comment]
Legend!

I'm working on a release this weekend where I'll want a summary row and table level is good enough for now.
Will start playing shortly.

I've held back on more suggestions+PR until I launch and get feedback from users.

Again, thank you.

**Comment by komarovalexander** [comment]
The group summary row released. It is available in v6.10.0: https://komarovalexander.github.io/ka-table/#/grouping-summary

**Comment by TimNZ** [comment]
@komarovalexander legend!  Thanks so much.
Migrating all my tables from react-table to KA.
